nodes/src/nodes/agent_rocketride/executor.py

```
@@ -176,125 +171,118 @@ def _result_size_label(result: Any) -> str:
+    n = min(_MAX_WORKERS, len(tagged))
+    results: List[Any] = [None] * len(tagged)
 
     with ThreadPoolExecutor(max_workers=n) as pool:
-        # Submit all calls and map each future back to its wave index
-        future_to_idx = {pool.submit(_run_one, call): i for i, call in enumerate(wave)}
+        future_to_idx = {pool.submit(_run_one, call): i for i, call in enumerate(tagged)}
         for future in as_completed(future_to_idx):
             idx = future_to_idx[future]
             try:
                 results[idx] = future.result(timeout=_TOOL_TIMEOUT_S)
             except Exception as exc:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)